### ISSUE TITLE
XNOR for ResNet

### ISSUE THREAD

**Issue by XinDongol** [jiecaoyu/XNOR-Net-PyTorch#21]
I was wondering whether there is XNOR for ResNet. Or can you give some hints about how to implement it?

**Comment by jxgu1016** [comment]
Also interested in XNOR for ResNet! @XinDongol have you implemented it yourself?

**Comment by 25b3nk** [comment]
Has anyone implemented XNOR for Resnet ?

**Comment by akhauriyash** [comment]
I do not mind implementing ResNet for this if there is interest. However, I will not be able to train it. 
I think this task mostly just involves changing the class AlexNet [here.](https://github.com/jiecaoyu/XNOR-Net-PyTorch/blob/master/ImageNet/networks/model_list/alexnet.py)

**Comment by 25b3nk** [comment]
I have implemented XNOR-net for ResNet. Basically took the `resnet.py` from torchvision and added the binary Convolution classes from this repo's `alexnet.py` and then changed the convolutions in `resnet.py` from *nn.Conv2D* to *BinConv2D*.

**Comment by XinDongol** [comment]
I have implemented ResNet-XNOR and reproduce result of xnor paper. I will release it soon when I have time.

**Comment by akhauriyash** [comment]
@XinDongol could you kindly share the python models directory code for the ResNet-xNOR with me?
Thanks!

**Comment by dorian95** [comment]
> I have implemented ResNet-XNOR and reproduce result of xnor paper. I will release it soon when I have time.

Could you please share it with me too? Thanks!

**Comment by appleleaves** [comment]
@XinDongol Would you like to share the python models directory code for the ResNet-XNOR or share the implementation idea?

**Comment by blueardour** [comment]
@XinDongol  Any update on the thread?

**Comment by legolas123** [comment]
I have implemented the xnornet ++ paper with resnet-18. You can disable alpha, beta, gamma(scale factors for activation) in model file(BinConv2d). Then you can have the xnornet for resnet-18. Btw xnornet++ is improvement over xnornet and my implementation reaches 60% accuracy for 1 bit activation and 1 bit width.
The link is https://github.com/legolas123/cv-tricks.com/tree/master/xnornet_plusplus
